ast/ddl.go

```diff
@@ -1807,6 +1808,17 @@ func (a AlterAlgorithm) String() string {
 	}
 }
 
+// ValidationType represents validation types of alter command
+type ValidationType byte
```

Comment:
https://dev.mysql.com/doc/refman/8.0/en/partitioning-management-exchange.html

> Optionally, you can append `WITH VALIDATION` or `WITHOUT VALIDATION`. … `WITH VALIDATION` is the default. 

I think a boolean indicating whether `WITHOUT VALIDATION` is specified is enough. Specifying `WITH VALIDATION` should be equivalent to not specifying it.

Reply:
that's ok. 

we would treat default as `WITH VALIDATION` in `AlterTableExchangePartition` spec. and when restoring, we just write `WITHOUT VALIDATION`, by doing this we would make no break change.